protocol/jsonlines/decoder.go

```diff
@@ -40,17 +40,41 @@ func NewDecoder(r io.Reader) Decoder {
 	return &decoder{r: lr}
 }
 
+func (d *decoder) discardPending() error {
+	// Read and discard everytihng until the next newline
+	for {
+		_, err := d.r.ReadSlice('\n')
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			} else if err == bufio.ErrBufferFull {
+				continue
+			} else {
+				return err
+			}
+		}
+	}
+}
+
 // Decode decodes the next line in the reader.
 // It does not check JSON for well-formedness before decoding, so in case of
 // error, the structure might be half-filled.
 func (d *decoder) Decode(v interface{}) error {
-	line, isPrefix, err := d.r.ReadLine()
-	if isPrefix {
-		return fmt.Errorf("buffer size exceeded")
-	}
+	var line []byte
+
+	for {
+		chunk, isPrefix, err := d.r.ReadLine()
+		if err != nil {
+			if !isPrefix {
+				d.discardPending()
```

Comment:
PS: is probably undefined behavious to do this with the bufio object once it has already failed, but I checked causing the failure and without the discard the next call would hang so it's probably the lesser evil.